### DISCUSSION TITLE
Funkcionalnosti

### DISCUSSION THREAD

**Discussion by marinjurisich** [marinjurisich/TimeKeeper#2] [Ideas]
**Aplikacija**
Time keeping

**Svrha**
Vođenje evidencije radnih sati
(vođenje projekta?)

**Korisnici (tipovi korisnika / uloge)**
- Zaposlenici - praćenje svojih sati
- Rukovoditelj (HR) - praćenje učinka odjela, praćenje svojih sati
- Privatne osobe (= HR u bazi) - učinkovitost rada na projektu

**Baza**
Korisnici - ime i prezime
(TODO)

**Funkcionalnosti (ovaj puta prave) - U obliku Storyja za projektni zadatak**
1. CRUD operacije / Baza podataka
2. Login, autentifikacija
3. Admin (HR acc) - registracija admin korisnika (+ firme), stvaranje korisnika
4. Korisnik - *DANAŠNJI DAN* - Clock in, clock out, odabir projekta (1 istovremeno), opis
5. Korisnik - *KALENDAR* - Prikaz radnih dana, blagdana, ..., slobodnih dana
6. Korisnik - *MODAL (Popup)* - Uređivanje jednog dana kalendara, sati
7. Kartični clock in/out - Demo korištenja APIja
8. Prikaz izvještaja (podataka) - radniku za sebe
9. Export podataka u excel
10. Popis korisnika
11. Mailanje - Slanje lozinke mailom
12. -----------------------

13. Importanje opisa sa GitHuba u određ. dan

**Comment by shtef21** [comment]
STARE
-------------

**Funkcionalnosti**
1. Login + autentifikacija
13. Naslovnica - Jednostavan graf koji sažima zadnjih X dana (trenutni mjesec?), satnica, ... (što još staviti)
14. Statistika za menadžere (HR) - Prosječna količina odrađenih sati po danu odjela / firme / tipa radnika, ... (smisliti)
15. Vanjski api na dodavanje sati - Npr clock in putem skeniranja kartice (simulirati prislanjanje kartice na neki način)
16. Izvoz podataka/statistike (u excel, za printanje, ...), uvoz podataka odjela i slično
17. Zasebni detaljni godišnji kalendar s prikazom radnih/neradnih dana, s prikazom već odrađenih dana, ...
18. Za određeni dan napisati opis što je sve učinjeno - opcija dodavanja slika/dokumenata kao priloga (ne direktno u tekst, nego u prilog, poput e-maila)
19. Opcija pullanja korisnikovih commitova i postavljanje kao posao odrađen na taj dan (u opis dana iz prethodne točke)
20. Mailanje - Potvrda korisnika, reset lozinke, prijava greške sustava rukovoditelju (HRu)


-----------------------
Popis funkcionalnosti koje bi mogli (ili trebali) implementirati. Dodajte svoje prijedloge ili komentirajte tuđe.

1. Login/sign up screen
21. Clock in/out button
22. Export data (to excel/csv)
23. Prikaz dosad odrađenih sati
24. Detaljan prikaz sati po danima
25. Edit clock in/out vremena
26. Kalendar s prikazom radnih/neradnih dana
